### PR TITLE
{Profile} `az login`: Add warning for username password flow

### PR DESCRIPTION
**Pull request by jiasli**
**Related command**
`az login`

**Description**
Add warning for username password flow according to https://github.com/Azure/azure-cli/issues/28252.

**Testing Guide**
```
az login --username xxx --password xxx
```

### CODE REVIEW

**Comment by yonzhan** [comment]
Profile

**Review by yonzhan — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/profile/custom.py [hunk 1]
```diff
@@ -41,6 +41,9 @@
     "[Warning] The login output has been updated. Please be aware that it no longer displays the full list of "
     "available subscriptions by default.\n")
 
+USERNAME_PASSWORD_DEPRECATION_WARNING = (
+    "User authentication with username and password has been deprecated. Please use interactive login instead.")
```
Comment (dcaro) [inline src/azure-cli/azure/cli/command_modules/profile/custom.py:45]:
```suggestion
    "Authentication with username and password in the command line is strongly discouraged. Use one of the recommended authentication methods based on your requirements. For more information go to https://learn.microsoft.com/cli/azure/authenticate-azure-cli".)
```
@dbradish-microsoft can you review the message?

**Review by dcaro — CHANGES_REQUESTED** [review]
@jiasli consider the updated message and change the title of this PR, we are not deprecating the username and password flow.

File: src/azure-cli/azure/cli/command_modules/profile/custom.py (hunk 1 above)
Comment (dbradish-microsoft) [reply src/azure-cli/azure/cli/command_modules/profile/custom.py:45]: @dcaro, Sound good to me.

**Comment by Alex-wdy** [comment]
I just to confirm that this information is only displayed in scenario where the customr uses username+password to login, and will not be displayed in other interactive scenarios, right?

File: src/azure-cli/azure/cli/command_modules/profile/custom.py (hunk 1 above)
Comment (isra-fel) [reply src/azure-cli/azure/cli/command_modules/profile/custom.py:45]: Let's replace the link with a fwlink as our common practice.

**Comment by jiasli** [comment]
> I just to confirm that this information is only displayed in scenario where the customr uses username+password to login, and will not be displayed in other interactive scenarios, right?

Correct.

File: src/azure-cli/azure/cli/command_modules/profile/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/profile/custom.py:45]: We usually use

> For more information, see ...

Comment (Jacekey23) [reply src/azure-cli/azure/cli/command_modules/profile/custom.py]: https://go.microsoft.com/fwlink/?linkid=2276314

**Review by bebound — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/profile/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/profile/custom.py]: @dcaro, updated as suggested.

**Review by dcaro — APPROVED** [review]
LGTM

**Review by yonzhan — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**